colorizer_data/tests/test_converter.py

```diff
@@ -0,0 +1,137 @@
+
+    expected_manifest = tmp_path / "dataset" / "manifest.json"
+    assert os.path.exists(expected_manifest)
+    manifest = {}
+    with open(expected_manifest, "r") as f:
+        manifest = json.load(f)
+
+    # Should not include data columns as features
+    assert manifest["features"] == [
+        {
+            "name": "Continuous Feature",
+            "key": "continuous_feature",
+            "type": "continuous",
+            "description": "",
+            "unit": "",
+            "min": 0.5,
+            "max": 0.8,
+            "data": "feature_0.json",
+        },
+        {
+            "name": "Discrete Feature",
+            "key": "discrete_feature",
+            "type": "discrete",
+            "description": "",
+            "unit": "",
+            "min": 0,
+            "max": 3,
+            "data": "feature_1.json",
+        },
+        {
+            "name": "Categorical Feature",
+            "key": "categorical_feature",
+            "type": "categorical",
+            "description": "",
+            "unit": "",
+            "min": 0,
+            "max": 2,
+            "categories": ["A", "B", "C"],
+            "data": "feature_2.json",
+        },
+    ]
+    assert os.path.exists(tmp_path / "dataset" / "feature_0.json")
+    assert os.path.exists(tmp_path / "dataset" / "feature_1.json")
+    assert os.path.exists(tmp_path / "dataset" / "feature_2.json")
+    validate_data(tmp_path / "dataset" / "feature_0.json", [0.5, 0.6, 0.7, 0.8])
+    validate_data(tmp_path / "dataset" / "feature_1.json", [0, 1, 2, 3])
+    validate_data(tmp_path / "dataset" / "feature_2.json", [0, 1, 2, 0])
+
+    assert manifest["frames"] == ["frame_0.png", "frame_1.png"]
+    assert os.path.exists(tmp_path / "dataset" / "frame_0.png")
+    assert os.path.exists(tmp_path / "dataset" / "frame_1.png")
+
+    assert manifest["tracks"] == "tracks.json"
+    assert os.path.exists(tmp_path / "dataset" / "tracks.json")
+    validate_data(tmp_path / "dataset" / "tracks.json", [1, 1, 2, 2])
+
+    assert manifest["centroids"] == "centroids.json"
+    assert os.path.exists(tmp_path / "dataset" / "centroids.json")
+    validate_data(
+        tmp_path / "dataset" / "centroids.json", [50, 50, 55, 60, 60, 70, 65, 75]
+    )
+
+    assert manifest["times"] == "times.json"
+    assert os.path.exists(tmp_path / "dataset" / "times.json")
+    validate_data(tmp_path / "dataset" / "times.json", [0, 1, 0, 1])
+
+
+"""
+TODO: Test additional edge cases
+- Frame generation
+  - override switch works
+  - detects change in number of objects
+  - detects removal of frames
+  - does not regenerate frames if they already exist
+- Handles missing centroid, outliers, or bounds data
+- Handles backdrop images via column
+- Handles backdrop images via dictionary
+"""
+
+
+def test_handles_renamed_columns(tmp_path):
+    pass
+
+
+def test_detects_missing_frames(tmp_path):
+    pass
+
+
+def test_handles_missing_data_columns(tmp_path):
```

Comment:
missing and or misnamed columns is a good test case just to make sure it errors out gracefully